data/types.go

```diff
@@ -36,17 +38,24 @@ type Signature struct {
 	Signature HexBytes `json:"sig"`
 }
 
-type Key struct {
-	Type       string   `json:"keytype"`
-	Scheme     string   `json:"scheme"`
-	Algorithms []string `json:"keyid_hash_algorithms,omitempty"`
-	Value      KeyValue `json:"keyval"`
+type PublicKey struct {
+	Type       string          `json:"keytype"`
+	Scheme     string          `json:"scheme"`
+	Algorithms []string        `json:"keyid_hash_algorithms,omitempty"`
+	Value      json.RawMessage `json:"keyval"`
 
 	ids    []string
```

Comment:
I am wondering why id is an array of string, not a string?

Reply:
I have no idea why it's a list -- I thought there was some context behind keys having multiple IDs but that doesn't seem to be the case in the Python implementation either. Should I open an issue to clean this up?

Reply:
Yup. An issue does that. -- If I were to guess, there might have been a reason to use multiple hashing algorithms there to diversify crypto, but that is a guess only.

Reply:
Yeah, this could be related to the old `key_hashing_algorithms` kerfuffle (TLDR: we allowed using SHA2-256 and/or SHA2-512 to compute different, collision-resistant keyids for the same key).

Reply:
https://github.com/theupdateframework/go-tuf/issues/158